### PR TITLE
[Nano] Support torch 2.0 in setup and action test

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

Support torch 2.0 in setup and action test

### 1. Why the change?

Support torch 2.0 in setup and action test to provide new installation option for user and support more 2.0 related features in the future.

### 2. User API changes

No change.

### 3. Summary of the change 

- [x] upgrade action default python version (in pytorch action test) from 3.7 to 3.8.16 as torch 2.0 only support python 3.8 and higher versions
- [x] torch2.0 requires inc >= 2.1 and inc 2.1 has a little different API with inc 2.0
- [x] ipex 2.0 update the check for AVX2 instruction sets, add related avx2_checker
`SystemExit: ERROR! Intel® Extension for PyTorch* only works on machines with instruction sets equal or newer than AVX2, which are not detected on the current machine.`
- [x] PyTorch Lightning scheduler issue
`pytorch_lightning.utilities.exceptions.MisconfigurationException: The provided lr scheduler `LambdaLR` doesn't follow PyTorch's LRScheduler API. You should override the `LightningModule.lr_scheduler_step` hook with your own logic if you are using a custom LR scheduler.`
we can't fix it except we upgrade pytorch lightning. Just skip related tests now.
We may upgrade pytorch lightning in next PR and update related code later.

### 4. How to test?
- [x] Unit test

### 5. What to do after this PR
- Upgrade PyTorch-Lightning to 2.0.0
- check or report the AVX2 behaviour of ipex

### CODE REVIEW

**Comment by qiuxin2012** [comment]
Please move your repo [support_torch2](https://github.com/intel-analytics/BigDL/tree/support_torch2) to your personal repository.

**Comment by rnwang04** [comment]
> Please move your repo [support_torch2](https://github.com/intel-analytics/BigDL/tree/support_torch2) to your personal repository.

will close this pr and branch
